### PR TITLE
Remove 'whitelist' and 'blacklist' terms

### PR DESCRIPTION
**Pull request by jmcphers**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/9621. 

### Approach

Replace the terms "whitelist" and "blacklist" in the codebase with more appropriate terms. 

### Automated Tests

None, no behavior change.

### QA Notes

The renamed options (named in the original issue) should be tested to be sure they still work as they did previously. 

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -41,6 +41,8 @@
 ### Misc
 
 * **BREAKING:** RStudio Desktop Pro only supports activation with license files (Pro #2300)
+* **BREAKING:** The `session-env-var-save-blacklist` option has been renamed to `session-ephemeral-env-vars`.
```
Comment (kevinushey) [inline NEWS.md:44]: Do you think it's worth supporting the old options? (I'm guessing these are used somewhat rarely, so perhaps that's okay?)

File: src/cpp/desktop/DesktopOptions.cpp [hunk 2]
```diff
@@ -347,15 +347,15 @@ void Options::setClipboardMonitoring(bool monitoring)
    settings_.setValue(QString::fromUtf8("clipboard.monitoring"), monitoring);
 }
 
-bool Options::ignoreGpuBlacklist() const
+bool Options::ignoreGpuExclusionList() const
 {
-   QVariant ignore = settings_.value(QStringLiteral("general.ignoreGpuBlacklist"), false);
+   QVariant ignore = settings_.value(QStringLiteral("general.ignoreGpuExclusionList"), false);
```
Comment (kevinushey) [inline src/cpp/desktop/DesktopOptions.cpp:352]: Does this imply that the user's old desktop options here would be lost on upgrade? (That's probably fine but wanted to double-check it's a conscious decision)

**Review by kevinushey — APPROVED** [review]
LGTM; just some questions about whether we're okay with not migrating some older options (I expect it's fine).

File: NEWS.md (hunk 1 above)
Comment (jmcphers) [reply NEWS.md:44]: Yes, these the `save-blacklist` was introduced for one customer's use case AFAIK, and I added the `view-whitelist` as an escape hatch but to my knowledge no one has actually needed to use it.

File: src/cpp/desktop/DesktopOptions.cpp (hunk 2 above)
Comment (jmcphers) [reply src/cpp/desktop/DesktopOptions.cpp:352]: Yep, that's what I meant to do. Losing this option just means that we will start using software rendering (the safer option) so a small handful of people will need to re-enable the option to use acceleration again.

**Pull request merged**